core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java

```diff
@@ -43,18 +49,28 @@ protected EmbeddedCacheManager createCacheManager() throws Exception {
    }
 
    public void testSimpleEvictionMaxEntries() throws Exception {
-      for (int i = 0; i < CACHE_SIZE*2; i++) {
-         cache.put("key-" + (i + 1), "value-" + (i + 1), 1, TimeUnit.MINUTES);
+      for (int i = 0; i < CACHE_SIZE * 2; i++) {
+         cache.put("key-" + (i + 1), "value-" + (i + 1));
       }
-      Thread.sleep(1000); // sleep long enough to allow the thread to wake-up
-      assert CACHE_SIZE >= cache.size() : "cache size too big: " + cache.size();
-      assert CACHE_SIZE == evictionListener.getEvictedEvents() : "eviction events count should be same with case size: " + evictionListener.getEvictedEvents();
+      assertEquals("cache size too big: " + cache.size(), CACHE_SIZE, cache.size());
+      assertEquals("eviction events count should be same with case size: " + evictionListener.getEvictedEvents(),
+            CACHE_SIZE, evictionListener.getEvictedEvents().size());
 
       for (int i = 0; i < CACHE_SIZE; i++) {
-         cache.put("key-" + (i + 1), "value-" + (i + 1), 1, TimeUnit.MINUTES);
+         cache.put("key-" + (i + 1), "value-" + (i + 1));
+      }
+      assertEquals("cache size too big: " + cache.size(), CACHE_SIZE, cache.size());
+      int targetSize;
+      if (getEvictionStrategy() == EvictionStrategy.LIRS) {
+         // Eviction count will be Size + (Size * .95) rounded up (since the first elements will be in resident blocks
+         // so they won't cause evictions to occur
+         targetSize = (int) Math.ceil(CACHE_SIZE + (CACHE_SIZE * .05));
```

Comment:
The variable name could be more specific, e.g. `expectedEvictions`.
Also, the comment says `Size + (Size * .95)` :)